src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -221,7 +221,13 @@ private SwiftCompileStep makeCompileStep(SourcePathResolverAdapter resolver) {
     ImmutableList.Builder<String> compilerCommand = ImmutableList.builder();
     compilerCommand.addAll(swiftCompiler.getCommandPrefix(resolver));
 
-    compilerCommand.add("-target", swiftTarget.getTriple());
+    String swiftTargetTriple = swiftTarget.getTriple();
+    // For arm64 simulator target, we need to add a "simulator" postfix to differentiate.
+    // Otherwise it'll be treated as targeting "ios" and would fail the compilation.
+    if (this.flavor.getName().contains("simulator") && this.flavor.getName().contains("arm64")) {
+      swiftTargetTriple += "-simulator";
```

Comment:
I'm fairly certain that we can append `-simulator` without checking the architecture. `x86_64-apple-ios12.0-simulator` is also valid.

Reply:
@qyang-nj, we need to check the architecture because the user might be building for "ios device".

Reply:
@xianwen Isn't this line `this.flavor.getName().contains("simulator")` guarantee it's building for simulator?

Reply:
@qyang-nj Ah, I see. I misunderstood you the first time.

I was trying to minimize the risk and impact of my changes. Maybe we can remove the check of architecture sometime later, but it's not a high priority issue and I wouldn't feel comfortable removing the checking without doing some extensive testing first.